src/main/java/com/spotify/reaper/service/SegmentRunner.java

```diff
@@ -260,33 +287,36 @@ boolean canRepair(RepairSegment segment, String keyspace, JmxProxy coordinator,
       repairRunner.updateLastEvent(msg);
       return false;
     }
+    
+    boolean gotMetricsForAllHosts = true;
 
     for (String hostName : allHosts) {
       LOG.debug("checking host '{}' for pending compactions and other repairs (can repair?)"
                 + " Run id '{}'", hostName, segment.getRunId());
       try (JmxProxy hostProxy = context.jmxConnectionFactory.connect(hostName)) {
-        int pendingCompactions = hostProxy.getPendingCompactions();
-        if (pendingCompactions > MAX_PENDING_COMPACTIONS) {
-          LOG.info("SegmentRunner declined to repair segment {} because of too many pending "
-                   + "compactions (> {}) on host \"{}\"", segmentId, MAX_PENDING_COMPACTIONS,
-              hostProxy.getHost());
-          String msg = String.format("Postponed due to pending compactions (%d)",
-              pendingCompactions);
-          repairRunner.updateLastEvent(msg);
-          return false;
+        Optional<HostMetrics> hostMetrics = getMetricsForHost(hostName, hostProxy);
+        if(!hostMetrics.isPresent()) {
+          gotMetricsForAllHosts = false;
         }
-        if (hostProxy.isRepairRunning()) {
-          LOG.info("SegmentRunner declined to repair segment {} because one of the hosts ({}) was "
-                   + "already involved in a repair", segmentId, hostProxy.getHost());
-          String msg = "Postponed due to affected hosts already doing repairs";
-          repairRunner.updateLastEvent(msg);
-          if (!busyHosts.get().contains(hostName)) {
-            LOG.warn("A host ({}) reported that it is involved in a repair, but there is no record "
-                + "of any ongoing repair involving the host. Sending command to abort all repairs "
-                + "on the host.", hostProxy.getHost());
-            hostProxy.cancelAllRepairs();
+        else {
+          int pendingCompactions = hostMetrics.get().getPendingCompactions();
+          if (pendingCompactions > MAX_PENDING_COMPACTIONS) {
+            LOG.info("SegmentRunner declined to repair segment {} because of too many pending "
+                     + "compactions (> {}) on host \"{}\"", segmentId, MAX_PENDING_COMPACTIONS,
+                hostProxy.getHost());
+            String msg = String.format("Postponed due to pending compactions (%d)",
+                pendingCompactions);
+            repairRunner.updateLastEvent(msg);
+            return false;
+          }
+          if (hostMetrics.get().hasRepairRunning()) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS134)